enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -700,6 +692,53 @@ func (c *FirecrackerContainer) LoadSnapshot(ctx context.Context, workspaceDirOve
 	if err != nil {
 		return status.WrapError(err, "Failed to initialize firecracker VM exec client")
 	}
+
+	if workspaceDirOverride != "" {
+		// If the snapshot is being loaded with a different workspaceFS
+		// then handle that now.
+		wsImgRelPath, err := c.createWorkspaceImage(ctx, workspaceDirOverride)
+		if err != nil {
+			return err
+		}
+		if err := c.hotSwapWorkspace(ctx, execClient, wsImgRelPath); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// createWorkspaceImage creates a new ext4 image from the action working dir
+// and returns the chroot-relative path to the created image.
+func (c *FirecrackerContainer) createWorkspaceImage(ctx context.Context, workspacePath string) (string, error) {
+	c.workspaceGeneration++
+	relativePath := fmt.Sprintf("%s.gen_%d.ext4", workspaceFSName, c.workspaceGeneration)
+	hostPath := filepath.Join(c.getChroot(), relativePath)
+	sizeBytes, err := disk.DirSize(workspacePath)
+	if err != nil {
+		return "", err
+	}
+	if err := ext4.DirectoryToImage(ctx, workspacePath, hostPath, sizeBytes+c.constants.DiskSlackSpaceMB*1e6); err != nil {
+		return "", err
+	}
+	c.workspaceFSPath = hostPath
+	return relativePath, nil
+}
+
+// hotSwapWorkspace unmounts the workspace drive from a running firecracker
+// container, updates the workspace block device to an ext4 image pointed to
+// by chrootRelativeImagePath, and re-mounts the drive.
+func (c *FirecrackerContainer) hotSwapWorkspace(ctx context.Context, execClient vmxpb.ExecClient, chrootRelativeImagePath string) error {
+	log.Infof("Hot-swapping guest workspace drive")
```

Comment:
Did you mean to leave this in? w/o additional container instance information it's not a very useful log line

Reply:
Woops, removed. Also added tracing